pkg/kubelet/kubelet_node_status.go

```diff
@@ -341,23 +341,32 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 }
 
 func (kl *Kubelet) setNodeStatusMachineInfo(node *api.Node) {
+	// Note: avoid blindly overwriting the capacity in case opaque
+	//       resources are being advertised.
+	if node.Status.Capacity == nil {
+		node.Status.Capacity = api.ResourceList{}
```

Comment:
@derekwaynecarr, do you still need to do some additional vetting of the affected Kubelet code? I checked for this during initial implementation, but If it would be helpful I could do an audit/writeup.